src/main/cpp/src/decimal_utils.cu

```diff
@@ -746,11 +753,15 @@ struct dec128_divider {
       auto const scale_divisor = pow_ten(n_shift_exp).as_128_bits();
 
       // The second divide gets the result into the scale that we care about and does the rounding.
-      auto const result = is_int_div ? integer_divide(first_div_result.quotient, scale_divisor)
-                            : divide_and_round(first_div_result.quotient, scale_divisor);
-
+      chunked256 result;
+      if constexpr (is_int_div) {
+        result = integer_divide(first_div_result.quotient, scale_divisor);
+        quotient_data[i] = result.as_64_bits();
+      } else {
+        result = divide_and_round(first_div_result.quotient, scale_divisor);
+        quotient_data[i] = result.as_128_bits();
+      }
```

Comment:
```suggestion
      if constexpr (is_int_div) {
        auto const result = integer_divide(first_div_result.quotient, scale_divisor);
        quotient_data[i] = result.as_64_bits();
      } else {
        auto const result = divide_and_round(first_div_result.quotient, scale_divisor);
        quotient_data[i] = result.as_128_bits();
      }
```

Reply:
result is being used outside of the if block. This suggestion will cause a compile error